### PR TITLE
[JENKINS-60866] Clean up build button column JS

### PR DESCRIPTION
**Pull request by daniel-beck**
See [JENKINS-60866](https://issues.jenkins-ci.org/browse/JENKINS-60866).

To support cleanup of the build button column, we need to be able to set IDs on icons, so that's being done here. Also deprecate `onclick` for being evil.

It would probably be cleaner to migrate `icon`/`svgIcon` to `MorphTagLibrary` to be able to add `data` attributes on it, but that's a larger change than necessary and it wouldn't get rid of the support element anyway; as `class` must only specify icons and cannot be used to add additional arbitrary class names.

### Proposed changelog entries

* Entry 1: Issue, Human-readable Text
* ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by Wadeck — APPROVED** [review]
Manually tested before and after, working fine in both cases.

**Review by oleg-nenashev — APPROVED** [review]
LGTM from the code PoV. Does it require a changelog entry?

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

File: core/src/main/resources/hudson/views/BuildButtonColumn/column.jelly
Comment (jglick) [inline core/src/main/resources/hudson/views/BuildButtonColumn/column.jelly:44]: Looks like we also need to update https://github.com/jenkinsci/jenkins/blob/af5a649a470f7fecff38de754df8ee10c9a17c62/core/src/main/resources/lib/hudson/project/configurable.jelly#L30 though that might require bigger changes https://github.com/jenkinsci/jenkins/blob/af5a649a470f7fecff38de754df8ee10c9a17c62/core/src/main/resources/lib/layout/task.jelly#L153

**Pull request merged**
